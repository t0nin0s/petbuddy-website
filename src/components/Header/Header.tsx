import React from "react";
import styled from 'styled-components'
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import IconButton from "@material-ui/core/IconButton";
import MenuIcon from "@material-ui/icons/Menu";
import MenuItem from "@material-ui/core/MenuItem";
import Menu from "@material-ui/core/Menu";

const linksArray: Array<string> = ['Home', 'Cat Services', 'Dog Services', 'Small Pet Services', 'Become a PBuddy', 'Contact Us']

const StyledAppBar = styled(AppBar)`
`
const StyledToolbar = styled(Toolbar)`
background-color: whitesmoke;
color: purple;
justify-content:space-between
`
const StyledTypography = styled(Typography)`
&& {
  font-size:2rem;
  font-weight:700
}
`
const StyledIconButton = styled(IconButton)`
 `
const StyledMenuIcon = styled(MenuIcon)`
  && {
  font-size:3rem;
  font-weight:700;
  color: purple;
  }
`
const StyledMenu = styled(Menu)`
margin-left: 50%;
margin-top: 17%;
width:150%;

`

export default function MenuAppBar() {
  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
  const open = Boolean(anchorEl);

  const handleMenu = (event: React.MouseEvent<HTMLElement>) => {
    setAnchorEl(event.currentTarget);
  };

  const handleClose = () => {
    setAnchorEl(null);
  };

  const ourServices = linksArray.map(service => {
    return <MenuItem onClick={handleClose}>{service}</MenuItem>
  })

  return (
    <div className="header-container">
      <StyledAppBar position="static">
        <StyledToolbar>
          <StyledTypography >
            PetBuddy
          </StyledTypography>
          <StyledIconButton
            onClick={handleMenu}
          >
            <StyledMenuIcon />
          </StyledIconButton>
          <StyledMenu
            open={open}
            onClose={handleClose}
          >
            {ourServices}
          </StyledMenu>
        </StyledToolbar>
      </StyledAppBar>
    </div>
  );
}