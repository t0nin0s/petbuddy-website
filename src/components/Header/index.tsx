import React from "react";
import styled from 'styled-components'
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import IconButton from "@material-ui/core/IconButton";
import MenuIcon from "@material-ui/icons/Menu";
import MenuItem from "@material-ui/core/MenuItem";
import Menu from "@material-ui/core/Menu";
import { mediaQueries } from '../../mediaqueries'

const linksArray: Array<string> = ['Cat Services', 'Dog Services', 'Small Pet Services', 'Become a PBuddy', 'Contact Us']

const StyledAppBar = styled(AppBar)``

const StyledToolbar = styled(Toolbar)`
  background-color: whitesmoke;
  color: purple;
  justify-content:space-between
`
const StyledTypography = styled(Typography)`
  font-size:2rem;
  font-weight:700
  `
const StyledIconButton = styled(IconButton)`
${mediaQueries("medium")`
  display:none;`}`

const StyledMenuIcon = styled(MenuIcon)`
  font-size:3rem;
  font-weight:700;
  color: purple;
  display:flex;
${mediaQueries("medium")`
  display:none;`}
`
const StyledMenu = styled(Menu)`
  margin-left: 3.5%;
`
const NavBarItems = styled.div`
display:none;
  ${mediaQueries("medium")`
    display:flex;
    flex-direction:row;
    flex-grow:1;
    justify-content: flex-end;
    color:#f48fb1;
    font-weight: 900;
    `}
`

export default function MenuAppBar() {
  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
  const open = Boolean(anchorEl);

  const handleMenu = (event: React.MouseEvent<HTMLElement>) => {
    setAnchorEl(event.currentTarget);
  };

  const handleClose = () => {
    setAnchorEl(null);
  };

  const ourServices = linksArray.map(service => {
    return <MenuItem onClick={handleClose}>{service}</MenuItem>
  })

  return (
    <div className="header-container">
      <StyledAppBar position="static">
        <StyledToolbar>
          <StyledTypography >
            PetBuddy
          </StyledTypography>
          <NavBarItems>{ourServices}</NavBarItems>
          <StyledIconButton
            onClick={handleMenu}>
            <StyledMenuIcon />
          </StyledIconButton>
          <StyledMenu
            anchorEl={anchorEl}
            getContentAnchorEl={null}
            anchorOrigin={{ vertical: "bottom", horizontal: "center" }}
            transformOrigin={{ vertical: "top", horizontal: "center" }}
            open={open}
            onClose={handleClose}
          >
            {ourServices}
          </StyledMenu>
        </StyledToolbar>
      </StyledAppBar>
    </div>
  );
}