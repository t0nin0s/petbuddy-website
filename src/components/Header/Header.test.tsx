import React from 'react'
import Header from './index'
import { render } from '@testing-library/react'

describe('Header component', () => {
  it('should render', () => {
    const wrapper = render(<Header />)
    expect(wrapper).toMatchSnapshot();
  })
  it('should output some categories', () => {
    const wrapper = 'Small Pet Services';
    expect(wrapper).toEqual('Small Pet Services')
  })
})