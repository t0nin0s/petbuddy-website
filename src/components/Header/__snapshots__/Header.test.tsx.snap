// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="header-container"
      >
        <header
          class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary sc-bdfBwQ mui-fixed MuiPaper-elevation4"
        >
          <div
            class="MuiToolbar-root MuiToolbar-regular sc-gsTCUz fWJebG MuiToolbar-gutters"
          >
            <p
              class="MuiTypography-root sc-dlfnbm dbIros MuiTypography-body1"
            >
              PetBuddy
            </p>
            <div
              class="sc-gKsewC eCvdcS"
            >
              <li
                aria-disabled="false"
                class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
                role="menuitem"
                tabindex="-1"
              >
                Cat Services
                <span
                  class="MuiTouchRipple-root"
                />
              </li>
              <li
                aria-disabled="false"
                class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
                role="menuitem"
                tabindex="-1"
              >
                Dog Services
                <span
                  class="MuiTouchRipple-root"
                />
              </li>
              <li
                aria-disabled="false"
                class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
                role="menuitem"
                tabindex="-1"
              >
                Small Pet Services
                <span
                  class="MuiTouchRipple-root"
                />
              </li>
              <li
                aria-disabled="false"
                class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
                role="menuitem"
                tabindex="-1"
              >
                Become a PBuddy
                <span
                  class="MuiTouchRipple-root"
                />
              </li>
              <li
                aria-disabled="false"
                class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
                role="menuitem"
                tabindex="-1"
              >
                Contact Us
                <span
                  class="MuiTouchRipple-root"
                />
              </li>
            </div>
            <button
              class="MuiButtonBase-root MuiIconButton-root sc-hKgILt dhwanL"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root sc-eCssSg kUykmG"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </header>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="header-container"
    >
      <header
        class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary sc-bdfBwQ mui-fixed MuiPaper-elevation4"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular sc-gsTCUz fWJebG MuiToolbar-gutters"
        >
          <p
            class="MuiTypography-root sc-dlfnbm dbIros MuiTypography-body1"
          >
            PetBuddy
          </p>
          <div
            class="sc-gKsewC eCvdcS"
          >
            <li
              aria-disabled="false"
              class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
              role="menuitem"
              tabindex="-1"
            >
              Cat Services
              <span
                class="MuiTouchRipple-root"
              />
            </li>
            <li
              aria-disabled="false"
              class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
              role="menuitem"
              tabindex="-1"
            >
              Dog Services
              <span
                class="MuiTouchRipple-root"
              />
            </li>
            <li
              aria-disabled="false"
              class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
              role="menuitem"
              tabindex="-1"
            >
              Small Pet Services
              <span
                class="MuiTouchRipple-root"
              />
            </li>
            <li
              aria-disabled="false"
              class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
              role="menuitem"
              tabindex="-1"
            >
              Become a PBuddy
              <span
                class="MuiTouchRipple-root"
              />
            </li>
            <li
              aria-disabled="false"
              class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
              role="menuitem"
              tabindex="-1"
            >
              Contact Us
              <span
                class="MuiTouchRipple-root"
              />
            </li>
          </div>
          <button
            class="MuiButtonBase-root MuiIconButton-root sc-hKgILt dhwanL"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root sc-eCssSg kUykmG"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </header>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
